bam_ampliconclip.c

```diff
@@ -145,6 +145,13 @@ int load_bed_file_multi_ref(char *infile, int get_strand, int sort_by_pos, khash
         }
 
         int cols_parsed = 0;
+
+// The construction of scanf_str based on the number of columns in use is
+// not compatible with -Wformat-nonliteral.  Short of lots of code duplication,
+// there is little we can do other than elide this warning for this piece of
+// code.  This is probably the lesser of two evils.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
```

Comment:
An easier solution here would be to always pass the full format list to `sscanf()` and then use the return value to see how many of them it managed to fill out.  Checking against `num_columns` should ensure that it hasn't been confused by whitespace other than tabs.